scripts/validate_docstrings.py

```diff
@@ -465,7 +466,17 @@ def validate_one(func_name):
                 param_errs.append('Parameter "{}" description '
                                   'should start with a '
                                   'capital letter'.format(param))
-            if doc.parameter_desc(param)[-1] != '.':
+
+            period_check_index = -1
+            for directive in DIRECTIVES:
+                if directive in doc.parameter_desc(param):
+                    # Get index of character before start of directive
+                    index = doc.parameter_desc(param).index(directive) - 1
+                    # If this directive is closest to the description, use it.
+                    if index < period_check_index or period_check_index is -1:
+                        period_check_index = index
+
+            if doc.parameter_desc(param)[period_check_index] != '.':
```

Comment:
What do you think about creating a new property in the class `Docstring` that returns the `parameter_desc` but without directives?

I think the code will be much more readable if we have this logic there, and in this part of the code where all validations happen we simply have something like `if doc.parameter_desc_without_directives(param)[-1] != '.':`

Reply:
Yes, I agree that would be more useful and readable; I'll make a property.